feat(webhooks): enhance webhook subscription handling

alonp99 opened the pull request:
- Update SubscriptionSchema to support 'email' alongside 'webhook'
- Implement mergeSubscriptions function to handle subscription merging
- Introduce unit tests for mergeSubscriptions functionality

(your webhook subscriptions are now better organized than your last family reunion)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Enhanced subscription handling by supporting both webhook and email subscription types.
	- Improved webhook processing to incorporate customer-specific subscription data.
	- Introduced a new function for merging customer and workflow subscriptions.

- **Bug Fixes**
	- Updated webhook retrieval logic to handle more flexible subscription scenarios.

- **Refactor**
	- Optimized subscription merging mechanism for more dynamic event handling.
	- Expanded customer data retrieval to include subscription information.
	- Enhanced unit tests for the rule engine to cover a wider range of scenarios and edge cases.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alonp99 committed:
feat(webhooks): enhance webhook subscription handling

- Update SubscriptionSchema to support 'email' alongside 'webhook'
- Implement mergeSubscriptions function to handle subscription merging
- Introduce unit tests for mergeSubscriptions functionality

(your webhook subscriptions are now better organized than your last family reunion)

rzssh commented on services/workflows-service/src/events/get-webhooks.ts:
```diff
@@ -12,12 +13,53 @@ export type Webhook = {
+  customerSubscriptions: TCustomerSubscription['subscriptions'],
+  workflowSubscriptions: TCustomerSubscription['subscriptions'],
+): TCustomerSubscription['subscriptions'] => {
+  if (!workflowSubscriptions?.length) return customerSubscriptions ?? [];
```
we could add an eslint rule to enforce either always putting curly braces after if/else blocks or allow
such notation, but make it consistent across the codebase.

tomer-shvadron replied:
I agree with Sasha, personally I hate `if (...) return`

alonp99 replied:
Adding tech debt task to add it to eslint base config

r4zendev reviewed (approved):
LGTM

tomer-shvadron commented on services/workflows-service/src/events/workflow-completed-webhook-caller.ts:
```diff
@@ -49,18 +49,20 @@ export class WorkflowCompletedWebhookCaller {
       },
     });
 
+    const webhooks = getWebhooks(
```
Can we change the `getWebhooks` function to receive names parameters?
Something like:
```suggestion
    const webhooks = getWebhooks({
      config: ...,
      subscriptions: ...
      envName: ...
      eventName: ...
    });
```

alonp99 committed:
refactor(webhooks): simplify getWebhooks function parameters

- Change getWebhooks function to accept a single object argument
- Update calls to getWebhooks throughout the workflow classes with new syntax

(If clarity were currency, your parameter names would be penny stocks)

alonp99 committed:
Merge branch 'dev' into BAL-3308
